src/leaflet_d3.js

```diff
@@ -58,15 +52,32 @@ L.CartoDBd3Layer = L.TileLayer.extend({
       provider: this.provider,
       map: map
     })
-    this._tileContainer.setAttribute('class', 'leaflet-zoom-animated leaflet-tile-container')
-    this._bgBuffer.setAttribute('class', 'leaflet-zoom-animated leaflet-tile-container')
-    this.tileLoader.on('tileAdded', this._renderTile, this)
-    this.tileLoader.on('tileRemoved', this._clearTile, this)
-    this._map.on({
-      'zoomanim': this._animateZoom,
-      'zoomend': this._endZoomAnim
-    }, this)
     this.tileLoader.loadTiles()
+    this.on('ready', function () {
```

Comment:
What happens if the layer receives this event multiple times? (e.g: if I do `layer.provider.setURL` multiple times)? Do we need to ensure the following bindings are "unbound" when a "ready" event is received (otherwise bindings will be appended to the existing ones)?

```      
      this.tileLoader.on('tileAdded', this._renderTile, this)
      this.tileLoader.on('tileRemoved', this._clearTile, this)
      this._map.on({
        'zoomanim': this._animateZoom,
        'zoomend': this._endZoomAnim
      }, this)
```